### ISSUE TITLE
ARM: issue with 'Set-AzureRmVirtualNetworkGatewayConnection'

### ISSUE THREAD

**Issue by romangromov** [Azure/azure-powershell#1327]
PS version 1.0.1

Question:
Here is description for `Set-AzureRmVirtualNetworkGatewayConnection`:

```
SYNTAX
    Set-AzureRmVirtualNetworkGatewayConnection [-Force] [-Profile [<AzureProfile>]] -VirtualNetworkGatewayConnection
    <PSVirtualNetworkGatewayConnection> [<CommonParameters>]
```

What does it mean **PSVirtualNetworkGatewayConnection**? It's not the same as `VirtualNetworkGatewayConnection` retrieved with `Get-AzureRmVirtualNetworkGatewayConnection`?

Right now I can't change existing connection, steps to repro:

```
PS C:\> $connection = Get-AzureRmVirtualNetworkGatewayConnection -Name LocalConnection -ResourceGroupName test-roman-gro
up
PS C:\> $connection.RoutingWeight = 888
PS C:\> Set-AzureRmVirtualNetworkGatewayConnection -VirtualNetworkGatewayConnection $connection
Set-AzureRmVirtualNetworkGatewayConnection :
Mapping types:
PSVirtualNetworkGatewayConnection -> VirtualNetworkGatewayConnection
Microsoft.Azure.Commands.Network.Models.PSVirtualNetworkGatewayConnection ->
Microsoft.Azure.Management.Network.Models.VirtualNetworkGatewayConnection
Destination path:
VirtualNetworkGatewayConnection
Source value:
Microsoft.Azure.Commands.Network.Models.PSVirtualNetworkGatewayConnection
At line:1 char:1
+ Set-AzureRmVirtualNetworkGatewayConnection -VirtualNetworkGatewayConnection $con ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Set-AzureRmVirt...tewayConnection], AutoMapperMappingException
    + FullyQualifiedErrorId : AutoMapper.AutoMapperMappingException,Microsoft.Azure.Commands.Network.SetAzureVirtualNe
   tworkGatewayConnectionCommand

PS C:\>
```

Please help me to figure this out, because I'm blocked and can't update an existing connection,
Thanks in advance.

**Comment by romangromov** [comment]
Any updates on this?

**Comment by MilanDasek** [comment]
I have the same issue - powershell 1.1.0

command New-AzureRMVirtualNetworkGatewayConnection

**Comment by andystumph** [comment]
I'm having a very similar issue. Is this a known bug?

PS C:\Windows\system32> $NGC[1].RoutingWeight = 50

PS C:\Windows\system32> Set-AzureRmVirtualNetworkGatewayConnection -VirtualNetworkGatewayConnection $NGC[1]
Set-AzureRmVirtualNetworkGatewayConnection : 'GatewayType' cannot be null.
At line:1 char:1
+ Set-AzureRmVirtualNetworkGatewayConnection -VirtualNetworkGatewayConnection $NGC ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureRmVirt...tewayConnection], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Network.SetAzureVirtualNetworkGatewayConnectionCommand

**Comment by cormacpayne** [comment]
@romangromov @MilanDasek if this happens to still be an issue for either of you, we will be shipping a fix for all AutoMapper mapping exceptions in the upcoming release of Azure PowerShell (version 5.0.0 - set to release on November 8th).

**Comment by cormacpayne** [comment]
This should be fixed in [Azure PowerShell 5.0.0](https://github.com/Azure/azure-powershell/releases/tag/v5.0.0-November2017). Please feel free to re-open this issue if you are still seeing this problem.
